lxd/db/cluster/constants.go

```diff
@@ -9,29 +9,68 @@ import (
 
 // Numeric type codes identifying different kind of entities.
 const (
-	TypeContainer = iota
-	TypeImage
-	TypeProfile
-	TypeProject
+	TypeContainer             = 0
+	TypeImage                 = 1
+	TypeProfile               = 2
+	TypeProject               = 3
+	TypeCertificate           = 4
+	TypeInstance              = 5
+	TypeInstanceBackup        = 6
+	TypeInstanceSnapshot      = 7
+	TypeNetwork               = 8
+	TypeNetworkACL            = 9
+	TypeNode                  = 10
+	TypeOperation             = 11
+	TypeStoragePool           = 12
+	TypeStorageVolume         = 13
+	TypeStorageVolumeBackup   = 14
+	TypeStorageVolumeSnapshot = 15
+	TypeWarning               = 16
 )
 
 // EntityNames associates an entity code to its name.
 var EntityNames = map[int]string{
-	TypeContainer: "container",
-	TypeImage:     "image",
-	TypeProfile:   "profile",
-	TypeProject:   "project",
+	TypeContainer:             "container",
+	TypeImage:                 "image",
+	TypeProfile:               "profile",
+	TypeProject:               "project",
+	TypeCertificate:           "certificate",
+	TypeInstance:              "instance",
+	TypeInstanceBackup:        "instance backup",
+	TypeInstanceSnapshot:      "instance snapshot",
+	TypeNetwork:               "network",
+	TypeNetworkACL:            "network acl",
+	TypeNode:                  "node",
+	TypeOperation:             "operation",
+	TypeStoragePool:           "storage pool",
+	TypeStorageVolume:         "storage volume",
+	TypeStorageVolumeBackup:   "storage volume backup",
+	TypeStorageVolumeSnapshot: "storage volume snapshot",
+	TypeWarning:               "warning",
 }
 
 // EntityTypes associates an entity name to its type code.
 var EntityTypes = map[string]int{}
 
 // EntityURIs associates an entity code to its URI pattern.
 var EntityURIs = map[int]string{
-	TypeContainer: "/" + version.APIVersion + "/containers/%s?project=%s",
-	TypeImage:     "/" + version.APIVersion + "/images/%s",
-	TypeProfile:   "/" + version.APIVersion + "/profiles/%s?project=%s",
-	TypeProject:   "/" + version.APIVersion + "/projects/%s",
+	TypeContainer:             "/" + version.APIVersion + "/containers/%s?project=%s",
+	TypeImage:                 "/" + version.APIVersion + "/images/%s",
```

Comment:
Not sure why this one isn't using ?project=%s.
Can you make a separate commit to correct it and track down where that format string is used to see if we need to pass it the project name at that spot?

Reply:
@stgraber `EntityURIs[TypeImage]` was used for creating [`projects_used_by_ref`](https://github.com/lxc/lxd/blob/master/lxd/db/cluster/update.go#L2355-L2368), but the latest change to `projects_used_by_ref` [hard-codes the URIs](https://github.com/lxc/lxd/blob/master/lxd/db/cluster/update.go#L118-L158).